src/pages/components/CertificateViewDetails.js

```diff
@@ -0,0 +1,220 @@
+import React from 'react'
+import styled from 'styled-components'
+
+import CertificateTimeInterval from './CertificateTimeInterval'
+
+import BgIconDateSVG from '../../assets/images/icon-date.svg'
+import BgIconTimeSVG from '../../assets/images/icon-time.svg'
+
+const hasValidDate = (d) => typeof d === 'string' && !isNaN(Date.parse(d))
+
+const hasValidEnergy = (e) => e !== null && e !== undefined && !isNaN(e)
+
+const hasValidSize = (s) => s !== null && s !== undefined && !isNaN(s)
+
+const hasValidEco2 = (e) => e !== null && e !== undefined && !isNaN(e) && e > 0
```

Comment:
I think this check is incorrect. An eCO2 value of `0` is valid. Also `isNaN` doesn't really check it is a number. `isFinite` along with `e >= 0` would be better

Reply:
Done